plugins/pulp_ostree/plugins/distributors/configuration.py

```diff
@@ -83,3 +96,42 @@ def get_repo_relative_path(repo, config):
     else:
         path = repo.repo_id
     return path
+
+
+def _check_for_relative_path_conflicts(repo_id, relative_path):
+    """
+    Check that a relative path does not conflict with existing distributors' relative paths.
+
+    :param repo_id: identifier of repository associated with the path to check
+    :type  repo_id: basestring
+    :param relative_path: relative path of the repository
+    :type  relative_path: basestring
+    :return error_messages: a list of validation errors
+    :rtype: list
+    """
+    current_url_pieces = [x for x in relative_path.split('/') if x]
+    matching_url_list = []
+    working_url = ''
+    for piece in current_url_pieces:
+        working_url = os.path.join(working_url, piece)
+        matching_url_list.append(working_url)
+        matching_url_list.append('/' + working_url)
+
+    # Search for all the sub urls as well as any url that would fall within the specified url.
+    # The regex here basically matches the a url if it starts with (optional preceding slash)
+    # the working url. Anything can follow as long as it is separated by a slash.
+    rel_url_match = Q(config__relative_path={'$regex': '^/?' + working_url + '(/.*|/?\z)'})
+    rel_url_in_list = Q(config__relative_path__in=matching_url_list)
```

Comment:
Let me make sure I have this right.  If the `relative_path` is: `/fedora/23/host`, the `matching_url_list` would contain:
'''
[
    fedora,
    /fedora,
    fedora/23,
    /fedora/23,
    fedora/23/host,
    /fedora/23/host,
'''
And the regex would be: ` '^/?fedora/23/host(/.*|/?\z)'} ` which would match on any `relative_path` starting with `fedora/23/host` (with optional leading /).

This that no other plugin publishes to the /fedora/23/host/*  tree.

Right.